### ISSUE TITLE
Split packages

### ISSUE THREAD

**Issue by ranisalt** [streamich/react-use#1019]
**Is your feature request related to a problem? Please describe.**
A somewhat follow up of #384, I would like to use this package with the minimum amount of dependencies pulled in.

**Describe the solution you'd like**
I should be able to use simple hooks such as `useEffectOnce` without pulling deps for other hooks such as `resize-observer-polyfill` for `useMeasure`.

**Describe alternatives you've considered**
It would be nice to separate each hook in a package using workspaces, it would also make it clear which hooks depend on each other through metadata. Instead of `yarn add react-use` I would use `yarn add @react-use/use-effect-once` or maybe a better naming scheme.

**Comment by chapterjason** [comment]
They should be defined as optional dependencies. Yarn and npm offers the `--no-optional` option for installing dependencies. Splitting packages would result in more `node_modules` folder hell.

**Comment by ranisalt** [comment]
@chapterjason I disagree on that. Scoped split packages generate a single directory under `node_modules`, and workspaces are very smart in avoiding duplication of dependencies. Optional dependencies [have a fallback plan if they are not available](https://classic.yarnpkg.com/en/docs/dependency-types/#toc-optionaldependencies), which is not the case: not having `resize-observer-polyfill` breaks `useMeasure`, thus it is not optional.

**Comment by andrewmclagan** [comment]
I believe splitting packages is pretty standard practice. e.g. we only use `useLocalStorage` and have to import the whole library... Not great. See Material UI and how they split it in the way the Op describes. Very simple process.

**Comment by ammmze** [comment]
I would also like to see these things split up similar to https://github.com/jaredLunde/react-hook. That would allow making breaking changes in an individual hook much more manageable without having to worry about whether or not you're gonna break other hook usages. And for people who still want an all-in-one package, we should be able to have a package that pulls everything in and re-exports the things.

**Comment by ranisalt** [comment]
And now someone already squatted in the `@react-use` scope in npmjs.com :shrug:

**Comment by gfox1984** [comment]
What a shame, I almost used this package... but at the end I decided against it as it bundles way too many things together... (I was only interested in lifecycle hooks)

**Comment by broofa** [comment]
> And now someone already squatted in the `@react-use` scope in npmjs.com 🤷

@ranisalt How can you tell?  (just curious how one can determine if a namespace has been taken on the NPM site, as that's not something I know how to do.)

**Comment by Rycochet** [comment]
Switch over to compiling via `vite` and it's easy to build multiple targets, with the esmodule version being both split (for direct imports) and tree-shakeable (for importing from the root) - I've done it on [lz-string ](https://github.com/pieroxy/lz-string)recently which is now also a pure TS project and wants similar things - have a look at the vite.config.mts file and corresponding entries in the package.json `exports` section!
